jobs/jobs/main.py

```diff
@@ -60,6 +60,32 @@ async def create_job(
     logger.info("Create job with job_params: %s", job_params)
     jw_token = token_data.token
 
+    # check if the job name is already taken
+    is_job_name_taken = db_service.is_job_exists_by_name(db, job_params.name)
+    if is_job_name_taken:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"The job name '{job_params.name}' is already being used",
+        )
+
+    # check if the given files exist in assets_files table
+    if len(job_params.files) > 0:
+        _, matched_file_ids_in_db = (
+            await utils.get_files_data_from_separate_files(
+                job_params.files, current_tenant, token_data.token
+            )
+        )
+
+        non_existing_file_ids = list(
+            set(job_params.files).difference(matched_file_ids_in_db)
+        )
```

Comment:
from a brief look it can be simplified to a function which takes ids of the files and fetch them from DB. If list of files returned by the function is less than the input - we are missing some of them